### ISSUE TITLE
ClearBit doesn't reset bits on time views

### ISSUE THREAD

**Issue by yuce** [pilosa/pilosa#1324]
This bug was originally reported at: https://groups.google.com/forum/#!topic/pilosa/zF68VGlkUXc

Create the schema and set a bit with timestamp, then clear that bit:
```
curl localhost:10101/index/index1 -d ''
curl localhost:10101/index/index1/frame/test1 -d '{"options": {"timeQuantum": "YMD"}}'

curl localhost:10101/index/index1/query -d 'SetBit(frame="test1", col=1, row=1, timestamp="2018-05-01T00:00")'
{"results":[true]}

curl localhost:10101/index/index1/query -d 'ClearBit(frame="test1", col=1, row=1, timestamp="2018-05-01T00:00")''
{"results":[true]}
```

### Expected behavior

The cleared bit should not exist:
```
curl localhost:10101/index/index1/query -d 'Range(frame="test1", row=1, start="2018-05-01T00:00", end="2018-05-02T00:00")'
{"results":[{"attrs":{},"bits":[]}]}
```

### Actual behavior

The bit still exists:
```
curl localhost:10101/index/index1/query -d 'Range(frame="test1", row=1, start="2018-05-01T00:00", 
{"results":[{"attrs":{},"bits":[1]}]}
```

### The problem

The problem is `executeClearBit` doesn't parse its `timestamp` argument like `SetBit` does and consequently `executeClearBitView` doesn't take a timestamp argument, and `f.ClearBit` is always called with a `nil` timestamp.

**Comment by travisturner** [comment]
The reason why this is not implemented is because if the same column has more than one (row, timestamp) value, then it's not possible to know which of the least significant time quantums are ok to delete.

Expanding on the example above:
```
SetBit(frame="test1", col=1, row=1, timestamp="2018-05-01T00:00")
SetBit(frame="test1", col=1, row=2, timestamp="2018-05-02T00:00")

ClearBit(frame="test1", col=1, row=1, timestamp="2018-05-01T00:00")
```
At this point it's not clear which of the time quantums to delete.
This ticket suggests deleting:
- 2018
- 2018-05
- 2018-05-01

but in doing so, two of the time quantums used by row 2 (2018 and 2018-05) would be deleted.

Another way to put it, supporting `ClearBit` with a timestamp only works if the data is guaranteed to only contain a single row for each column. So we need to decide if that's something we want to enforce.

**Comment by htrendev** [comment]
There is also the case of using `ClearBit()` without a timestamp
```
ClearBit(frame="test1", col=1, row=1)
```
on bits with timestamps. This also leaves the timestamped bit set as far as `Range()` queries are concerned.

Maybe it's safe to say that `ClearBit()`, when used without a timestamp parameter, should clear the bit as well as all its timestamps (even if more than one)?

This still won't provide a way to selectively clear a specific timestamp in case there are more than one, but it will at least allow for a bit to be cleared completely, so that `Range()` and `Bitmap()` queries agree on its state.

**Comment by jaffee** [comment]
fixed in 1.0
